### PR TITLE
Add Footer

### PR DESCRIPTION
**Pull request by msssk**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue: #11 
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

* Add a Footer module
    * The footer is not rendered by default. If the grid property `footerChildren` is specified then the footer will be rendered, and contain the provided children.
* Add a Footer example page
* Add Footer unit tests

### CODE REVIEW

**Comment by dylans** [comment]
This PR appears to have a bunch of commits from February, probably needs a rebase @msssk. Never mind, I see your note about what it is based off of now.

**Comment by msssk** [comment]
I didn't realize https://github.com/pottedmeat/dgrid/tree/initial wasn't based off master. It has some useful things in it, like an example page. I think we should get a foundation for example pages into master ASAP as it greatly facilitates development. It's a bit awkward developing off of current master since there's no environment to render what you're working on and see it working.

**Comment by msssk** [comment]
This PR is now based on master.

**Review by agubler — COMMENTED** [review]
Have you thought about using the new harness from @dojo/test-extras for the unit tests.

**Comment by dylans** [comment]
> Have you thought about using the new harness from @dojo/test-extras for the unit tests.

I don't think that's really been announced to the team as ready for general use yet, has it? I think for dgrid that would be awesome, but perhaps as a separate PR?

**Review by dylans — APPROVED** [review]
With the caveat that we will want to switch the test harness under a separate PR, I think this is great and should land.

File: src/Footer.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { v } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { RegistryMixin, RegistryMixinProperties }  from '@dojo/widget-core/mixins/Registry';
```
Comment (pottedmeat) [inline src/Footer.ts]: I think there's an extra space before `from`

File: src/styles/footer.m.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+@import './shared/variables.css';
+
+.footer {
+	background-color: var(--header-background-color);
+	bottom: 0;
+	position: absolute;
```
Comment (pottedmeat) [inline src/styles/footer.m.css]: Grid styling won't require things to be absolutely positioned, etc. the way they were in dgrid1. We should be able to remove bottom/position rules (see https://github.com/dojo/dgrid/issues/13)

**Review by pottedmeat — CHANGES_REQUESTED** [review]

**Review by pottedmeat — APPROVED** [review]

**Pull request merged**